File: ios/BluetoothModule.m
//
//  BluetoothModule.m
//  G1_GlassesApp
//


#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(BluetoothModule, NSObject)

RCT_EXTERN_METHOD(startScan:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(stopScan:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(connectToDevice:(NSString *)deviceName resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(disconnectFromGlasses:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(sendData:(NSString *)data lr:(NSString *)lr resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(enableMicrophone:(BOOL)enable resolve:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/BluetoothModule.swift
//
//  BluetoothModule.swift
//  Runner
//
//  Created by Aljaz on 2025/03/10.
//

import Foundation
import CoreBluetooth
import React

@objc(BluetoothModule)
class BluetoothModule: NSObject {
    private let bluetoothManager = BluetoothManager(channel: FlutterMethodChannel())

    @objc func startScan(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        bluetoothManager.startScan { result in
            resolve(result)
        }
    }

    @objc func stopScan(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        bluetoothManager.stopScan { result in
            resolve(result)
        }
    }

    @objc func connectToDevice(_ deviceName: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        bluetoothManager.connectToDevice(deviceName: deviceName) { result in
            resolve(result)
        }
    }

    @objc func disconnectFromGlasses(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        bluetoothManager.disconnectFromGlasses { result in
            resolve(result)
        }
    }

    @objc func sendData(_ data: String, lr: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        let params: [String: Any] = ["data": data, "lr": lr]
        bluetoothManager.sendData(params: params)
        resolve(nil)
    }

    @objc func enableMicrophone(_ enable: Bool, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        bluetoothManager.enableMicrophone(enable) { result in
            resolve(result)
        }
    }
}
